packages/wdio-devtools-service/src/utils.js

```diff
@@ -145,11 +150,22 @@ export function isBrowserVersionLower (caps, minVersion) {
  * @param   {string|*}      version chromedriver version like `78.0.3904.11` or just `78`
  * @return  {number|*}              either major version, ex `78`, or whatever value is passed
  */
-export function getChromeMajorVersion (version) {
+export function getBrowserMajorVersion (version) {
     let majorVersion = version
     if (typeof version === 'string') {
         majorVersion = Number(version.split('.')[0])
         majorVersion = isNaN(majorVersion) ? version : majorVersion
     }
     return majorVersion
 }
+
+/**
+ * check if browser is supported based on caps.browserName and caps.version
+ * @param {object} caps capabilities
+ */
+export function isBrowserSupported(caps) {
+    if ((caps.browserName && !(caps.browserName.toLowerCase() in SUPPORTED_BROWSERS_AND_MIN_VERSIONS))
+    || (caps.browserName && isBrowserVersionLower(caps, SUPPORTED_BROWSERS_AND_MIN_VERSIONS[caps.browserName.toLowerCase()])))
+        return false
```

Comment:
```suggestion
    if (
        !caps.browserName ||
        !(caps.browserName.toLowerCase() in SUPPORTED_BROWSERS_AND_MIN_VERSIONS) ||
        isBrowserVersionLower(caps, SUPPORTED_BROWSERS_AND_MIN_VERSIONS[caps.browserName.toLowerCase()]))
    ) {
        return false
    }
```

Reply:
@christian-bromann If I set `microsoftedge` as the browserName I start getting this error:

```
ERROR @wdio/devtools-service: Couldn't connect to chrome: net::ERR_ABORTED at chrome://version
    at Frame.<anonymous> (/Users/pjcalvo/Documents/repos-hoodoo/shc-automation-wdio/node_modules/puppeteer-core/lib/helper.js:94:19)
    at Page.goto (/Users/pjcalvo/Documents/repos-hoodoo/shc-automation-wdio/node_modules/puppeteer-core/lib/Page.js:485:53)
    at Page.<anonymous> (/Users/pjcalvo/Documents/repos-hoodoo/shc-automation-wdio/node_modules/puppeteer-core/lib/helper.js:95:27)
    at DevToolsDriver.navigateTo (/Users/pjcalvo/Documents/repos-hoodoo/shc-automation-wdio/node_modules/devtools/build/commands/navigateTo.js:13:14)
    at Browser.wrappedCommand (/Users/pjcalvo/Documents/repos-hoodoo/shc-automation-wdio/node_modules/devtools/build/devtoolsdriver.js:92:47)
    at findCDPInterface (/Users/pjcalvo/Documents/repos-opensource/webdriverio/packages/wdio-devtools-service/src/utils.js:37:26)
    at DevToolsService.before (/Users/pjcalvo/Documents/repos-opensource/webdriverio/packages/wdio-devtools-service/src/index.js:43:41)
    at Array.map (<anonymous>)
[0-0] 2020-06-24T13:06:38.358Z ERROR @
```

Which is thrown by this block here:
```js
/**
     * check if interface is part of goog:chromeOptions value
     */
    const chromeOptions = global.browser.capabilities['goog:chromeOptions']
    if (chromeOptions && chromeOptions.debuggerAddress) {
        const [host, port] = chromeOptions.debuggerAddress.split(':')
        return { host, port: parseInt(port, 10) }
    }

    /**
     * otherwise look into chrome flags
     */
    await global.browser.url('chrome://version') // here is where it breaks
    const cmdLineTextElem = await global.browser.$('#command_line')
    const cmdLineText = await cmdLineTextElem.getText()
    let port = parseInt(cmdLineText.match(RE_DEVTOOLS_DEBUGGING_PORT_SWITCH)[1], 10)
```

This is what I said in a couple of comments above that currently the devtools service is mostly supporting chrome. If you think it is worth it I can invest time on checking the browsertype first and figuring out the right version url, but I think if we add more support for other chromium browsers like Opera then we will end up changing this again.


@christian-bromann I changed the if logic as suggested (which I think is a better approach). Just be noticed that this changes the If logic: browserName was able to be sent as empty and it returned `isSupported` now it will be the opposite. I added a new service test to validate this.

Reply:
🤔 did you push the change? It seems to be as before.